runtime/lib/backends/nccl/providers/op_registration.cc

```diff
@@ -26,16 +30,72 @@ namespace cuda {
 
 void RegisterNCCLOps(KernelRegistry *registry) {
   registry->Register(
-      "NCCLRecv_f32",
+      "nccl.Recv",
       [](const brt::OpKernelInfo &info) -> std::shared_ptr<brt::OpKernel> {
-        auto kernel = std::shared_ptr<OpKernel>(new cuda::Recv<float>(info));
+        auto memrefType = info.GetOperation()
+                              ->getOperand(0)
+                              .getType()
+                              .cast<mlir::MemRefType>();
+        std::shared_ptr<brt::OpKernel> kernel;
+        if (memrefType.getElementType() ==
+            mlir::Float32Type::get(info.GetOperation()->getContext()))
+          kernel = std::shared_ptr<OpKernel>(new cuda::Recv<float>(info));
         return kernel;
       });
 
   registry->Register(
-      "NCCLSend_f32",
+      "nccl.Send",
       [](const brt::OpKernelInfo &info) -> std::shared_ptr<brt::OpKernel> {
-        auto kernel = std::shared_ptr<OpKernel>(new cuda::Send<float>(info));
+        auto memrefType = info.GetOperation()
+                              ->getOperand(0)
+                              .getType()
+                              .cast<mlir::MemRefType>();
+        std::shared_ptr<brt::OpKernel> kernel;
+        if (memrefType.getElementType() ==
```

Comment:
Is it necessary to dispatch by dtype at this point? Why not dispatch it directly?